### ISSUE TITLE
Dynamic data can be reused as extra static data

### ISSUE THREAD

**Issue by AntoinePassemiers** [janursa/geneRNI#25]
This is somewhat related to issue #23 but assuming we have a multivariate model that predicts (static) gene expression and (dynamic) gene expression changes, I think there is no reason not to use the time series as extra static data to augment the data set size for the static part of the model.

**Comment by janursa** [comment]
but for the same reason that you use multivariate model, i.e. the nature of this data is different or at least how we simulate them is different. using time series data for static data would be still valuable?

**Comment by janursa** [comment]
i see. how do you account for the decay coeff in the static model if you use time series? or you simply ignore that it's time series?

**Comment by AntoinePassemiers** [comment]
You don't use the decay coef, the target `y` is made of the raw expression values

**Comment by janursa** [comment]
true, but by treating time series data as static data, you remove the importance of time in the cellular reaction

**Comment by AntoinePassemiers** [comment]
We are not losing anything since time series are also processed as time series with the decay coef and so on

**Comment by AntoinePassemiers** [comment]
I'll try to make it clearer. My suggestion is to build bivariate models that jointly estimate the expression of the target gene at time t based on expression of regulators at time t, and the change between t and t+h based on expression of regulators at time t (of course the input is the same). Let's call the predicted pair `(y, dy_dt)`.

- If we only have static data, then only `y` would be considered when training the model.
- If we only have time series, then both `y` and `dy_dt` are relevant since time series can also be treated in a static fashion.
- If we have both static data and time series, then both `y` and `dy_dt` will be considered for samples originating from time series, and only `y` will be considered for the ones coming from static data.

So the model should be able to handle missing values in the targets. For neural networks that's straightforward (probably not with scikit-learn though), for linear models you simply split it into two separate models, etc.

**Comment by janursa** [comment]
thanks for the detailed layout. i will think about it tomorrow in the morning when my brain is clear again haha

**Comment by janursa** [comment]
In the third scenario, where both static data and time series data are present, the model for predicting y will consider both y and dy_dt for samples from time series data, and only y for samples from static data. However, since the model is independent of the dy model, it's important to consider the factor of time when training the model for predicting y. This can be done by incorporating time information, i.e. lags, into the inputs of the model

**Comment by janursa** [comment]
btw, i created this response using chatGPT :D

**Comment by AntoinePassemiers** [comment]
To incorporate time information, we have the dy_dt labels. Since we consider a bivariate model, knowledge will be shared between the static and dynamic parts of the model.

For example, if we build a multi layer perceptron with 2 outputs y and dy_dt, the loss function (e.g. MSE) over dy_dt will update the weights in the hidden layer, which are involved in the process of predicting **both** y and dy_dt. 

Another example with multi-task lasso (which is available in sklearn, but not sure it handles missing values), the linear coefficients are constrained to be similar across the 2 tasks. So the knowledge accumulated while training on dy_dt will guide the model toward learning similar coefficients for predicting y, and vice versa.

So again, time information is taken into account by the whole model through dy_dt

**Comment by janursa** [comment]
sounds intriguing. maybe we can also talk about this in our next meeting

**Comment by AntoinePassemiers** [comment]
The library now supports arbitrary multi-output regression. The main difference is that from now on `y` is 2-d.
